### ISSUE TITLE
Configuration problems with v0.9 (ECONNREFUSED)

### ISSUE THREAD

**Issue by dhirschfeld** [jupyterhub/jupyterhub#1869]
I'm trying to update my below configuration to no use deprecated attributes. The below configuration works:
```
IPADDRESS = public_ips()[0]
c.JupyterHub.ip = IPADDRESS
c.JupyterHub.hub_ip = '0.0.0.0'
c.JupyterHub.hub_port = 8081
c.JupyterHub.hub_connect_ip = IPADDRESS
```
**Output:**
```
[I 2018-05-07 15:35:18.032 JupyterHub app:981] Loading cookie_secret from C:\JupyterHub\jupyterhub_cookie_secret
[I 2018-05-07 15:35:18.097 JupyterHub proxy:429] Generating new CONFIGPROXY_AUTH_TOKEN
[I 2018-05-07 15:35:18.186 JupyterHub app:1145] Not using whitelist. Any authenticated user will be allowed.
[I 2018-05-07 15:35:18.236 JupyterHub app:1742] Hub API listening on http://0.0.0.0:8081/hub/
[I 2018-05-07 15:35:18.240 JupyterHub app:1744] Private Hub API connect url http://IPADDRESS:8081/hub/
[W 2018-05-07 15:35:18.243 JupyterHub proxy:481] Running JupyterHub without SSL.  I hope there is SSL termination happening somewhere else...
[I 2018-05-07 15:35:18.243 JupyterHub proxy:483] Starting proxy @ http://IPADDRESS:8000/
15:35:18.566 - info: [ConfigProxy] Proxying http://IPADDRESS:8000 to (no default)
15:35:18.569 - info: [ConfigProxy] Proxy API at http://127.0.0.1:8001/api/routes
15:35:18.778 - info: [ConfigProxy] 200 GET /api/routes 
[I 2018-05-07 15:35:18.952 JupyterHub proxy:299] Checking routes
[I 2018-05-07 15:35:18.953 JupyterHub proxy:368] Adding default route for Hub: / => http://IPADDRESS:8081
15:35:18.957 - info: [ConfigProxy] Adding route / -> http://IPADDRESS:8081
15:35:18.959 - info: [ConfigProxy] 201 POST /api/routes/ 
[I 2018-05-07 15:35:19.749 JupyterHub app:1799] JupyterHub is now running at http://IPADDRESS:8000
[I 2018-05-07 15:35:31.038 JupyterHub log:158] 302 GET / -> /hub (@10.200.18.81) 2.00ms
[I 2018-05-07 15:35:31.266 JupyterHub log:158] 302 GET /hub -> /hub/login (@10.200.18.81) 1.00ms
[I 2018-05-07 15:35:32.331 JupyterHub log:158] 302 GET /hub/login -> /user/dhirschf/ (dhirschf@10.200.18.81) 92.00ms
[I 2018-05-07 15:35:32.335 JupyterHub log:158] 302 GET /user/dhirschf/ -> /hub/user/dhirschf/ (@10.200.18.81) 2.00ms
...server starts up fine.
```
----

The updated config below results in an `ECONNREFUSED` error:
```
IPADDRESS = public_ips()[0]
c.JupyterHub.bind_url = 'http://0.0.0.0:8000'
c.JupyterHub.hub_bind_url = 'http://0.0.0.0:8081'
c.JupyterHub.hub_connect_ip = IPADDRESS
```
**Output:**
```
[I 2018-05-07 15:26:22.318 JupyterHub app:981] Loading cookie_secret from C:\JupyterHub\jupyterhub_cookie_secret
[I 2018-05-07 15:26:22.377 JupyterHub proxy:429] Generating new CONFIGPROXY_AUTH_TOKEN
[I 2018-05-07 15:26:22.451 JupyterHub app:1145] Not using whitelist. Any authenticated user will be allowed.
[I 2018-05-07 15:26:22.494 JupyterHub app:1742] Hub API listening on http://0.0.0.0:8081
[I 2018-05-07 15:26:22.496 JupyterHub app:1744] Private Hub API connect url http://IPADDRESS:63171/hub/
[W 2018-05-07 15:26:22.499 JupyterHub proxy:481] Running JupyterHub without SSL.  I hope there is SSL termination happening somewhere else...
[I 2018-05-07 15:26:22.499 JupyterHub proxy:483] Starting proxy @ http://0.0.0.0:8000/
15:26:22.806 - info: [ConfigProxy] Proxying http://0.0.0.0:8000 to (no default)
15:26:22.809 - info: [ConfigProxy] Proxy API at http://127.0.0.1:8001/api/routes
15:26:28.532 - info: [ConfigProxy] 200 GET /api/routes 
15:26:28.728 - info: [ConfigProxy] Adding route / -> http://IPADDRESS:63171
[I 2018-05-07 15:26:28.724 JupyterHub proxy:299] Checking routes
[I 2018-05-07 15:26:28.725 JupyterHub proxy:368] Adding default route for Hub: / => http://IPADDRESS:63171
15:26:28.730 - info: [ConfigProxy] 201 POST /api/routes/ 
[I 2018-05-07 15:26:29.521 JupyterHub app:1799] JupyterHub is now running at http://0.0.0.0:8000
15:26:36.135 - error: [ConfigProxy] 503 GET / connect ECONNREFUSED IPADDRESS:63171
15:26:37.142 - error: [ConfigProxy] Failed to get custom error page Error: connect ECONNREFUSED IPADDRESS:63171
    at Object.exports._errnoException (util.js:1020:11)
    at exports._exceptionWithHostPort (util.js:1043:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1105:14)
...Service Unavailable web page
```

**Comment by dhirschfeld** [comment]
I'll continue to try and get to the bottom of it, but if anything jumps out at anyone I'll gladly take any suggestions!

**Comment by dhirschfeld** [comment]
### Note:
I'm running JupyterHub on win64/py36

**Comment by dhirschfeld** [comment]
I'm not sure where the 63171 port is coming from in the bad config:

**Good:**
```
[I 2018-05-07 15:35:18.952 JupyterHub proxy:299] Checking routes
[I 2018-05-07 15:35:18.953 JupyterHub proxy:368] Adding default route for Hub: / => http://IPADDRESS:8081
15:35:18.957 - info: [ConfigProxy] Adding route / -> http://IPADDRESS:8081
```

**Bad:**
```
15:26:28.728 - info: [ConfigProxy] Adding route / -> http://IPADDRESS:63171
[I 2018-05-07 15:26:28.724 JupyterHub proxy:299] Checking routes
[I 2018-05-07 15:26:28.725 JupyterHub proxy:368] Adding default route for Hub: / => http://IPADDRESS:63171
```

**Comment by minrk** [comment]
hub_bind_url wasn't having the intended effect. Fixed in #1871. Thanks for testing!

**Comment by dhirschfeld** [comment]
No worries! See also #1870 which I think is required (but haven't tested yet)
